leaf-ui/js/backendComm.js

```diff
@@ -97,37 +97,22 @@ function responseFunc(isGetNextSteps, response){
 					console.log("in backendcomm, saving all next state results");
 					 open_analysis_viewer();
 			} else {
-				var resultsString = JSON.stringify(results);
-				 console.log(JSON.stringify(results)); 
-				 savedAnalysisData.singlePathResult = results;
-				 analysisResult.assignedEpoch = results.assignedEpoch;
-				 analysisResult.timePointPath = results.timePointPath;
-				 analysisResult.timePointPathSize = results.timePointPathSize;
-				 analysisResult.elementList = results.elementList;
-				 analysisResult.allSolution = results.allSolution;
-				 analysisRequest.previousAnalysis = analysisResult;
-				 console.log("previousAnalysis");
-				 console.log(analysisRequest.previousAnalysis);
-				 displayAnalysis(results);
-
-				 analysisResult.colorVis = new EVO(results.elementList);
-				 analysisResult.isPathSim = true;
-				 analysisResult.colorVis.singlePathResponse(results.elementList);
-
-				 // Save analysisResult to the corresponding analysis configuration object
-				 currAnalysisConfig.addAnalysis(analysisResult);
+				console.log(JSON.stringify(results)); 
+				savedAnalysisData.singlePathResult = results;
+				analysisResult = convertToAnalysisResult(results);
+				displayAnalysis(results);
+
+				 // Save result to the corresponding analysis configuration object
+				 currAnalysisConfig.addResult(convertToAnalysisResult(results));
```

Comment:
any reason for not simplifying? no need to call `convertToAnalysisResult()` twice
```suggestion
				 currAnalysisConfig.addResult(analysisResult);
```

Reply:
No reason at all, thanks for the suggestion!